### DISCUSSION TITLE
what action needed for image limit excedding

### DISCUSSION THREAD

**Discussion by Adnan0061** [vercel/vercel#8665] [Help] [answered]
My image optimization limit exceeds maybe because of storybook build, 
so what action is required now to get out of the situation? 

**situation images are given below**

![Screenshot 2022-10-03 135940](https://user-images.githubusercontent.com/48943431/193528420-7a413801-27c1-4bf3-9a2c-066a11e060f9.jpg)

![Screenshot 2022-10-03 135404](https://user-images.githubusercontent.com/48943431/193528814-6051bd73-ce46-467d-9447-06c9a2efb3b6.jpg)

**Comment by amyegan** [comment, marked as answer]
The [Image Optimization Limits](https://vercel.com/docs/concepts/image-optimization/limits-and-pricing#) restrict Hobby plan projects to 1000 source images, as you saw. You can work around this limit by skipping image optimization where appropriate, reducing the number of source images in your project, or upgrading to a Pro plan to increase your image optimization limit to 5000 images with the option to pay for additional images beyond that limit.

_Note from [image optimization documentation](https://vercel.com/docs/concepts/image-optimization#source-images):_
> Next.js projects that configure the [loader](https://nextjs.org/docs/basic-features/image-optimization#loader) config or [unoptimized](https://nextjs.org/docs/api-reference/next/image#unoptimized) prop will opt out of Vercel Image Optimization in favor of a 3rd party image provider. Therefore, the [Usage](https://vercel.com/docs/concepts/limits/usage) page will show 0 Source Images each billing period.
>
> Next.js projects that configure the [domains](https://nextjs.org/docs/basic-features/image-optimization#domains) config will likely see a high number of Source Images, especially if users of the website can upload arbitrary images that will be optimized. This is because the use of the domains property will mean images from external sites may also be optimized.

**Reply by Adnan0061** [reply]
thanks a lot. but what should I do now to avoid a ban? will deleting the custom domain help?

**Reply by amyegan** [reply]
The correct course of action is, of course, dependent upon exactly what you are trying to do. But if you are okay with skipping image optimization for remote images, then you could remove the [domains config](https://nextjs.org/docs/basic-features/image-optimization#domains) by not using an absolute URL for Image `src` prop. Using a [custom domain](https://vercel.com/docs/concepts/projects/domains) for your project in general should not have any affect on image optimization limits.

If you want to skip image optimization for all images in your project, you can set [`images.unoptimized`](https://nextjs.org/docs/api-reference/next/image#unoptimized) to `true` in your `next.config.js`. 

The important thing is to get the number of optimized images below 1000 (or upgrade to a Pro plan to increase the limit). Which method you choose is up to you. Do whatever makes sense to you for your project 😄

**Reply by Adnan0061** [reply]
I will do the configuration. 
but as for the current month, it's already crossed the limit should I do anything so that my account doesn't get banned this is the real question I am asking.

**Reply by amyegan** [reply]
For this month, simply adding the `unoptimized` prop is the most certain way to prevent yourself from increasing the overage. As long as you make the change, there is room for a bit of grace. If you continue to push further over the limit you may get blocked.
